app/scripts/lib/ens-ipfs/setup.js

```diff
@@ -43,10 +43,11 @@ function setupEnsIpfsResolver ({ provider, getIpfsGateway }) {
     const ipfsGateway = getIpfsGateway()
     extension.tabs.update(tabId, { url: `loading.html` })
     let url = `https://app.ens.domains/name/${name}`
+    const completeSearch = (search || '') + (!search || search.indexOf('?') === -1 ? '?' : '&') + 'ensd=' + name
```

Comment:
In addition to checking for existing query params, it might be a good idea to ensure `ensd` itself isn't already set. If we include it twice, it might not be parsed the way we expect.

Not sure whether we'd want to leave alone a pre-existing `ensd` parameter or overwrite it though 🤔

Reply:
In query parameters, it is allowed to have multiple vars with the same name: it is one of the allowed methods to implicitly declare that the var is an array, as you can see here in the first example. https://medium.com/raml-api/arrays-in-query-params-33189628fa68

As this PR would just be a help to users who could potentially need it, we cannot imagine the scenario they are running: one of those could be that they need ensd as an array var.

So I would not bother to check that ensd is a unique parameter, that is the user's business, we are only concerned with passing it on.

What do you think?